File: src/main/java/pl/yalgrin/playnite/simplesync/dto/filter/FilterPresetSettingsDTO.java
package pl.yalgrin.playnite.simplesync.dto.filter;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.io.Serial;
import java.io.Serializable;

@EqualsAndHashCode
@Data
@NoArgsConstructor
@SuperBuilder
public class FilterPresetSettingsDTO implements Serializable {
    @Serial
    private static final long serialVersionUID = 1511413271560442002L;

    @JsonProperty("UseAndFilteringStyle")
    private boolean useAndFilteringStyle;
    @JsonProperty("IsInstalled")
    private boolean installed;
    @JsonProperty("IsUnInstalled")
    private boolean uninstalled;
    @JsonProperty("Hidden")
    private boolean hidden;
    @JsonProperty("Favorite")
    private boolean favorite;
    @JsonProperty("Name")
    private String name;
    @JsonProperty("Version")
    private String version;

    @JsonProperty("ReleaseYear")
    private StringItemPropertiesDTO releaseYear;

    @JsonProperty("Genre")
    private IdItemPropertiesDTO genre;
    @JsonProperty("Platform")
    private IdItemPropertiesDTO platform;
    @JsonProperty("Publisher")
    private IdItemPropertiesDTO publisher;
    @JsonProperty("Developer")
    private IdItemPropertiesDTO developer;
    @JsonProperty("Category")
    private IdItemPropertiesDTO category;
    @JsonProperty("Tag")
    private IdItemPropertiesDTO tag;
    @JsonProperty("Series")
    private IdItemPropertiesDTO series;
    @JsonProperty("Region")
    private IdItemPropertiesDTO region;
    @JsonProperty("Source")
    private IdItemPropertiesDTO source;
    @JsonProperty("AgeRating")
    private IdItemPropertiesDTO ageRating;
    @JsonProperty("Library")
    private IdItemPropertiesDTO library;
    @JsonProperty("CompletionStatuses")
    private IdItemPropertiesDTO completionStatuses;
    @JsonProperty("Feature")
    private IdItemPropertiesDTO feature;

    @JsonProperty("UserScore")
    private IntItemPropertiesDTO userScore;
    @JsonProperty("CriticScore")
    private IntItemPropertiesDTO criticScore;
    @JsonProperty("CommunityScore")
    private IntItemPropertiesDTO communityScore;
    @JsonProperty("LastActivity")
    private IntItemPropertiesDTO lastActivity;
    @JsonProperty("RecentActivity")
    private IntItemPropertiesDTO recentActivity;
    @JsonProperty("Added")
    private IntItemPropertiesDTO added;
    @JsonProperty("Modified")
    private IntItemPropertiesDTO modified;
    @JsonProperty("PlayTime")
    private IntItemPropertiesDTO playTime;
    @JsonProperty("InstallSize")
    private IntItemPropertiesDTO installSize;
}

File: src/main/kotlin/pl/yalgrin/playnite/simplesync/library/dto/filter/FilterPresetSettingsDTO.kt
package pl.yalgrin.playnite.simplesync.library.dto.filter

import com.fasterxml.jackson.annotation.JsonProperty
import java.io.Serial
import java.io.Serializable

data class FilterPresetSettingsDTO(
    @param:JsonProperty("useAndFilteringStyle")
    @get:JsonProperty("useAndFilteringStyle")
    @field:JsonProperty("useAndFilteringStyle")
    var useAndFilteringStyle: Boolean = false,
    @param:JsonProperty("isInstalled")
    @get:JsonProperty("isInstalled")
    @field:JsonProperty("isInstalled")
    var isInstalled: Boolean = false,
    @param:JsonProperty("isUninstalled")
    @get:JsonProperty("isUninstalled")
    @field:JsonProperty("isUninstalled")
    var isUninstalled: Boolean = false,
    @param:JsonProperty("isHidden")
    @get:JsonProperty("isHidden")
    @field:JsonProperty("isHidden")
    var isHidden: Boolean = false,
    @param:JsonProperty("isFavorite")
    @get:JsonProperty("isFavorite")
    @field:JsonProperty("isFavorite")
    var isFavorite: Boolean = false,
    var name: String? = null,
    var version: String? = null,
    var releaseYear: StringItemPropertiesDTO? = null,
    var genre: IdItemPropertiesDTO? = null,
    var platform: IdItemPropertiesDTO? = null,
    var publisher: IdItemPropertiesDTO? = null,
    var developer: IdItemPropertiesDTO? = null,
    var category: IdItemPropertiesDTO? = null,
    var tag: IdItemPropertiesDTO? = null,
    var series: IdItemPropertiesDTO? = null,
    var region: IdItemPropertiesDTO? = null,
    var source: IdItemPropertiesDTO? = null,
    var ageRating: IdItemPropertiesDTO? = null,
    var library: IdItemPropertiesDTO? = null,
    var completionStatuses: IdItemPropertiesDTO? = null,
    var feature: IdItemPropertiesDTO? = null,
    var userScore: IntItemPropertiesDTO? = null,
    var criticScore: IntItemPropertiesDTO? = null,
    var communityScore: IntItemPropertiesDTO? = null,
    var lastActivity: IntItemPropertiesDTO? = null,
    var recentActivity: IntItemPropertiesDTO? = null,
    var added: IntItemPropertiesDTO? = null,
    var modified: IntItemPropertiesDTO? = null,
    var playTime: IntItemPropertiesDTO? = null,
    var installSize: IntItemPropertiesDTO? = null
) : Serializable {
    companion object {
        @Serial
        private const val serialVersionUID = 1511413271560442002L
    }
}